packages/lib/modules/code-review/rate-providers/useRateProvider.ts

```diff
@@ -0,0 +1,25 @@
+import { useRateProvidersQuery } from './useRateProvidersQuery'
+import { Address } from 'viem'
+
+export function useRateProvider() {
+  const { data: rateProvidersCodeReviewData, isLoading, error } = useRateProvidersQuery()
+
+  function getRateProvider(rateProviderAddress: Address) {
+    return rateProvidersCodeReviewData?.find(
+      rateProvider => rateProvider.address.toLowerCase() === rateProviderAddress.toLowerCase()
+    )
+  }
```

Comment:
I'm having trouble selecting a token w/o a rate provider because of error but if suggestion above is viable would render this moot

```
useRateProvider.ts:9 Uncaught TypeError: Cannot read properties of undefined (reading 'toLowerCase')
    at useRateProvider.ts:9:82
    at Array.find (<anonymous>)
    at o (useRateProvider.ts:8:41)
    at isConstantRateProvider (useRateProvider.ts:14:5)
    at q (ChoosePoolTokens.tsx:55:36)
    at onTokenSelect (ChoosePoolTokens.tsx:63:41)
    at I (TokenSelectModal.tsx:51:5)
    at onClick (TokenSelectList.tsx:123:48)